src/compas_timber/connections/half_lap.py

```diff
@@ -0,0 +1,152 @@
+        self.features = []
+
+    @property
+    def data(self):
+        data_dict = {
+            "beam_a": self.beam_a.key,
+            "beam_b": self.beam_b.key,
+        }
+        data_dict.update(Joint.data.fget(self))
+        return data_dict
+
+    @data.setter
+    def data(self, value):
+        Joint.data.fset(self, value)
+        self.beam_a_key = value["beam_a"]
+        self.beam_b_key = value["beam_b"]
+
+    @property
+    def joint_type(self):
+        return "Half-Lap"
+
+    @property
+    def beams(self):
+        return [self.beam_a, self.beam_b]
+
+    def _cutplane(self):
+        # Sets the Plane where the Cut will be
+        centerline_a = self.beam_a.centerline
+        centerline_b = self.beam_b.centerline
+        plane_a = Plane.from_frame(self.beam_a.faces[3])
+        plane_b = Plane.from_frame(self.beam_b.faces[3])
+        int_a, int_b = intersection_line_line_3D(centerline_a, centerline_b, self.max_distance)
+        int_a, _ = int_a
+        int_b, _ = int_b
+        point_cut = Point(*midpoint_point_point(int_a, int_b))  # If Centerline don't intersect
+        if self.cut_plane_choice == None or "A":
+            plane = plane_a
+        if self.cut_plane_choice == "B":
+            plane = plane_b
+        plane_cut = Plane(point_cut, plane[1])
+        return plane_cut
+
+    @staticmethod
+    def _create_polyhedron(plane_top, plane_bottom, lines_vertical):
+        # Step 1: Get 8 Intersection Points from 2 Planes and 4 Lines
+        int_points = []
+        for i in lines_vertical:
+            point_top = intersection_line_plane(i, plane_top)
+            point_bottom = intersection_line_plane(i, plane_bottom)
+            point_top = Point(*point_top)
+            point_bottom = Point(*point_bottom)
+            int_points.append(point_top)
+            int_points.append(point_bottom)
+        # Step 2: Create a hexahedron with 6 Faces from the 8 Points
+        return Polyhedron(
+            int_points,
+            [
+                [1, 3, 5, 7],  # top
+                [0, 2, 4, 6],  # bottom
+                [1, 3, 2, 0],  # left
+                [3, 5, 4, 2],  # back
+                [5, 7, 6, 4],  # right
+                [7, 1, 0, 6],  # front
+            ],
+        )
+
+    @staticmethod
+    def _mesh_to_brep(negative_polyhedron_beam_a, negative_polyhedron_beam_b):
+        # Show Polyhedrons
+        negative_polyhedron_beam_a = Artist(negative_polyhedron_beam_a).draw()
+        negative_polyhedron_beam_b = Artist(negative_polyhedron_beam_b).draw()
+
+        # Breps from Polyhedrons
+        negative_brep_beam_a = RhinoBrep.CreateFromMesh(rs.coercemesh(negative_polyhedron_beam_a), True)
+        negative_brep_beam_b = RhinoBrep.CreateFromMesh(rs.coercemesh(negative_polyhedron_beam_b), True)
+        return negative_brep_beam_a, negative_brep_beam_b
+
+    def _create_negative_volumes(self):
+        # Get Planes from Beams
+        plane_a = Plane.from_frame(self.beam_a.faces[3])
```

Comment:
Please double-check if these face assignments can generalize (see previous comment on orientation of z dir etc.)